internal/catalogmetadata/client/client.go

```diff
@@ -31,14 +75,75 @@ func (c *Client) Bundles(ctx context.Context) ([]*catalogmetadata.Bundle, error)
 		return nil, err
 	}
 	for _, catalog := range catalogList.Items {
-		channels, err := fetchCatalogMetadata[catalogmetadata.Channel](ctx, c.cl, catalog.Name, declcfg.SchemaChannel)
+		channels := []*catalogmetadata.Channel{}
+		bundles := []*catalogmetadata.Bundle{}
+
+		// TODO: When a catalogd release containing https://github.com/operator-framework/catalogd/pull/168
+		// is made this should be updated in favor of parsing `Catalog.Status` for the
+		// appropriate URL for fetching catalog contents.
+		req, err := http.NewRequestWithContext(ctx, http.MethodGet, fmt.Sprintf(catalogdOnClusterURLTemplate, c.baseURL, catalog.Name), nil)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("error forming request: %s", err)
 		}
 
-		bundles, err := fetchCatalogMetadata[catalogmetadata.Bundle](ctx, c.cl, catalog.Name, declcfg.SchemaBundle)
+		cacheData, ok := c.cache[catalog.Name]
+		if ok {
+			req.Header.Add("If-Modified-Since", cacheData.LastModifiedTime)
```

Comment:
Can you use information on the `CatalogSource` object to detect whether your client's cache is out-of-date, definitively? If not, is it possible to expose such information from `catalogd` to facilitate it?

Reply:
`Catalog` has `status.resolvedSource`, which will tell you the SHA resolution of whatever was in the spec. In theory, we can keep track of this resolved ref and only query if we have to cache or there's a mismatch.

Reply:
I believe this has been resolve as of https://github.com/operator-framework/operator-controller/pull/411/commits/fe924c2c2f9c686d3720bb930d42f01590409df1 so marking it as such. Feel free to unresolve if this is inaccurate